modules/entity_version_history/src/Entity/HistoryTabSettings.php

```diff
@@ -0,0 +1,205 @@
+    return $this->target_entity_type_id . '.' . $this->target_bundle;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTargetEntityTypeId(): string {
+    return $this->target_entity_type_id;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTargetBundle(): string {
+    return $this->target_bundle;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTargetField(): string {
+    return $this->target_field;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setTargetBundle(string $target_bundle): ConfigEntityInterface {
+    $this->target_bundle = $target_bundle;
+
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setTargetField(string $target_field): ConfigEntityInterface {
+    $this->target_field = $target_field;
+
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function preSave(EntityStorageInterface $storage): void {
+    $this->id = $this->id();
+    parent::preSave($storage);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function calculateDependencies(): ConfigEntityInterface {
+    parent::calculateDependencies();
+
+    // Create dependency on the bundle.
+    $entity_type = \Drupal::entityTypeManager()->getDefinition($this->target_entity_type_id);
+    $bundle_config_dependency = $entity_type->getBundleConfigDependency($this->target_bundle);
+    $this->addDependency($bundle_config_dependency['type'], $bundle_config_dependency['name']);
+
+    return $this;
+  }
+
+  /**
+   * Loads a HistoryTabSettings config based on the entity type and bundle.
+   *
+   * @param string $entity_type_id
+   *   ID of the entity type.
+   * @param string $bundle
+   *   Bundle name.
+   *
+   * @return null|$this
+   *   The HistoryTabSettings config entity if one exists. Otherwise, NULL.
+   */
+  public static function loadByEntityTypeBundle(string $entity_type_id, string $bundle): ?EntityInterface {
+    if ($entity_type_id === NULL || $bundle === NULL) {
+      return NULL;
+    }
+    return \Drupal::entityTypeManager()->getStorage('entity_version_history_settings')->load($entity_type_id . '.' . $bundle);
+  }
+
+  /**
+   * Loads all HistoryTabSettings config based on the entity type.
+   *
+   * @param string $entity_type_id
+   *   ID of the entity type.
+   *
+   * @return array
+   *   Array of HistoryTabSettings config entities if there is any.
+   */
+  public static function loadByEntityType(string $entity_type_id): array {
+    $configs = [];
+    if ($entity_type_id === NULL) {
+      return $configs;
+    }
+
+    $definition = \Drupal::entityTypeManager()->getDefinition($entity_type_id);
+    $bundles = \Drupal::entityTypeManager()->getStorage($definition->getBundleEntityType())->loadMultiple();
+    foreach ($bundles as $bundle_name => $bundle) {
+      $configs[$entity_type_id . '.' . $bundle_name] = \Drupal::entityTypeManager()->getStorage('entity_version_history_settings')->load($entity_type_id . '.' . $bundle_name);
```

Comment:
theoretically, we could use \Drupal\Core\Config\ConfigFactory::listAll(). This method support prefix.